db/migrate/20161118150205_add_user_email_fingerprint.rb

```diff
@@ -17,13 +17,15 @@ def down
     rename_column :users, :email_plain, :email
     decrypt_user_emails
     change_column_null :users, :email, false
+    add_index :users, :email, unique: true
     remove_column :users, :email_fingerprint
     remove_column :users, :encrypted_email
   end
 
   def encrypt_user_emails
     User.where(encrypted_email: '').each do |user|
-      ee = EncryptedEmail.new_from_email(user.email)
+      email_address = user.email.present? ? user.email : user.id.to_s
```

Comment:
the latter. the code will break if the db is rolled back to before `encrypted_email` exists but the index needs to be unique.